src/common/k8s-api/kube-object.ts

```diff
@@ -265,20 +513,31 @@ export class KubeObject<Metadata extends KubeObjectMetadata = KubeObjectMetadata
     return this.metadata.resourceVersion;
   }
 
+  getDescriptor() {
```

Comment:
I would avoid naming something `descriptor`. It very important meaning in javascript.

Reply:
Good point, will change.